read-fonts/src/tables/postscript/scale.rs

```diff
@@ -0,0 +1,541 @@
+}
+
+impl Default for HintParams {
+    fn default() -> Self {
+        Self {
+            blues: Blues::default(),
+            other_blues: Blues::default(),
+            family_blues: Blues::default(),
+            family_other_blues: Blues::default(),
+            // See <https://learn.microsoft.com/en-us/typography/opentype/spec/cff2#table-16-private-dict-operators>
+            blue_scale: Fixed::from_f64(0.039625),
+            blue_shift: Fixed::from_i32(7),
+            blue_fuzz: Fixed::ONE,
+            language_group: 0,
+        }
+    }
+}
+
+/// Entries that we parse from the Top DICT that are required to support
+/// charstring evaluation.
+#[derive(Default)]
+struct ScalerTopDict<'a> {
+    charstrings: Option<Index<'a>>,
+    font_dicts: Option<Index<'a>>,
+    fd_select: Option<FdSelect<'a>>,
+    private_dict_range: Option<Range<usize>>,
+    var_store: Option<ItemVariationStore<'a>>,
+}
+
+impl<'a> ScalerTopDict<'a> {
+    fn new(table_data: &'a [u8], top_dict_data: &'a [u8], is_cff2: bool) -> Result<Self, Error> {
+        let mut items = ScalerTopDict::default();
+        for entry in dict::entries(top_dict_data, None) {
+            match entry? {
+                dict::Entry::CharstringsOffset(offset) => {
+                    items.charstrings = Some(Index::new(
+                        table_data.get(offset..).unwrap_or_default(),
+                        is_cff2,
+                    )?);
+                }
+                dict::Entry::FdArrayOffset(offset) => {
+                    items.font_dicts = Some(Index::new(
+                        table_data.get(offset..).unwrap_or_default(),
+                        is_cff2,
+                    )?);
+                }
+                dict::Entry::FdSelectOffset(offset) => {
+                    items.fd_select = Some(FdSelect::read(FontData::new(
+                        table_data.get(offset..).unwrap_or_default(),
+                    ))?);
+                }
+                dict::Entry::PrivateDictRange(range) => {
+                    items.private_dict_range = Some(range);
+                }
+                dict::Entry::VariationStoreOffset(offset) if is_cff2 => {
+                    items.var_store = Some(ItemVariationStore::read(FontData::new(
+                        // IVS is preceded by a 2 byte length
+                        table_data.get(offset + 2..).unwrap_or_default(),
+                    ))?);
+                }
+                _ => {}
+            }
+        }
+        Ok(items)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::FontRef;
+
+    fn check_blues(blues: &Blues, expected_values: &[(f64, f64)]) {
+        for (i, blue) in blues.values().iter().enumerate() {
+            let expected = expected_values[i];
+            assert_eq!(blue.0, Fixed::from_f64(expected.0));
+            assert_eq!(blue.1, Fixed::from_f64(expected.1));
+        }
+    }
+
+    #[test]
+    fn read_noto_serif_display() {
```

Comment:
Suggest naming based on what the test is examining, for example `read_cff_static` would tell me more than this name and a pattern would emerge as you go on to `read_cff2_static`, etc.

Reply:
Renamed the tests as suggested